README.md

```diff
@@ -1,36 +1,121 @@
-# Cairo VM in Typescript
+# TypeScript Cairo VM
+
+<div align="center">
+  <h1><code>TypeScript Cairo VM</code></h1>
+
+<strong>An implementation of the Cairo VM in TypeScript, focusing on
+education</strong>
+
+[Github](https://github.com/kkrt-labs/cairo-vm-ts)
+
+<sub>Built with 🥕 by <a href="https://twitter.com/KakarotZkEvm">KKRT
+Labs</a></sub>
+
+</div>
+
+> ⚠️ This project is in its genesis, undergoing fast-paced development. It is
+> not suitable for production yet. Expect frequent breaking changes.
 
 ## What is Cairo
 
-<!-- TODO: fill this section -->
+Cairo stands for CPU AIR "o" (like insp"o", conv"o", 🤔).
 
-Cairo stands for CPU AIR "o" (like insp"o", conv"o", 🤔). See
-[Cairo whitepaper](https://eprint.iacr.org/2021/1063) for more information.
+It is a framework (a machine, an assembly and a language) which allows writing
+provable programs without having to understand the underneath ZK-technology.
+
+A program written in Cairo (or Cairo Zero) is compiled and then executed on a
+Cairo VM which produces traces to be used for the STARK proof generation.
+
+See [resources](#resources) for more information.
 
 ## Why this
 
-There are currently three Cairo VM implementations:
+### Implementation Diversity
+
+There are currently five other Cairo VM implementations:
 
-- the original
-  [python implementation](https://github.com/starkware-libs/cairo-lang) by
+<!-- Should I add the cairo vm in gs ? and 'oriac', a toy vm in rust -->
+
+- Reference (original)
+  [implementation in Python](https://github.com/starkware-libs/cairo-lang) by
   Starkware (prod)
-- the new [rust implementation](https://github.com/lambdaclass/cairo-vm) by
+- New [implementation in Rust](https://github.com/lambdaclass/cairo-vm) by
   Lambda Class (prod)
-- a new [go implementation](https://github.com/lambdaclass/cairo-vm_in_go) by
+- A [Go implementation](https://github.com/lambdaclass/cairo-vm_in_go), by
   Lambda Class (dev)
+- Another [Go implementation](https://github.com/NethermindEth/cairo-vm-go) by
+  Nethermind (dev)
+- A
+  [Zig implementation](https://github.com/keep-starknet-strange/ziggy-starkdust),
+  by Community (dev)
+- A [C++ implementation](https://github.com/lambdaclass/cairo-vm.c) by Lambda
+  Class (dev)
+
+The Lambda Class alt-implementations comes with a detailed guide
+([Go](https://github.com/lambdaclass/cairo-vm_in_go/blob/main/README.md#documentation),
+[C++](https://github.com/lambdaclass/cairo-vm.c?tab=readme-ov-file#documentation))
+on how they built their Cairo VM. It gives insights into the overall Cairo VM
+but is incomplete and rather specific to language details.
+
+Why would you have different implementations of the same program in multiple
+languages? For **implementation diversity**. As more implementations provide
```

Comment:
also mentioned the fact that the EVM reference implementation is *not* one of the client implementation but a dedicated EF supported python project https://github.com/ethereum/execution-specs/